tests/test_Config.cpp

```diff
@@ -92,7 +92,15 @@ class Temp {
 
 private:
     static rwfs::path getRandomFilePath(const rwfs::path &dirname) {
```

Comment:
I would replace the generation of random paths with an implementation similar to the following:
```
    static std::string gen_random(size_t len) {
        constexpr std::string_view alphanum =
            "0123456789"
            "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
            "abcdefghijklmnopqrstuvwxyz";
        std::default_random_engine reng(std::random_device{}());
        std::uniform_int_distribution<size_t> dist(0u, alphanum.size());
        std::string res;
        res.reserve(len);
        std::generate_n(std::back_inserter(res), len, [&]() {
            return alphanum[dist(reng)];
        });
        return res;
    }
    static rwfs::path getRandomFilePath(const rwfs::path &dirname) {
        std::string name = "openrw_test_" + gen_random(16);
        return dirname / name;
    }
```
This is more c++ :smiley:

Reply:
I guess my suggestion contained a bug.
`std::random_device{}()` can return the same seed if a non-deterministic source is not available.
Try this:
```
    static std::default_random_engine& getRandomEngine() {
        static std::default_random_engine defaultRandomEngine = std::default_random_engine(std::random_device{}());
        return defaultRandomEngine;
    }
    static std::string gen_random(size_t len) {
        constexpr std::string_view alphanum =
            "0123456789"
            "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
            "abcdefghijklmnopqrstuvwxyz";
        std::uniform_int_distribution<size_t> dist(0u, alphanum.size());
        std::string res;
        res.reserve(len);
        std::generate_n(std::back_inserter(res), len, [&]() {
            return alphanum[dist(getRandomEngine())];
        });
        return res;
    }
```